Problem2.py

```diff
@@ -0,0 +1,49 @@
+"""
+// Time Complexity : O(n), 
+// Space Complexity : O(1)
+// Did this code successfully run on Leetcode : NO, unable to figure out the mistake. Help needed.
```

Comment:
This code works, Check this out

class Solution:


    def __init__(self):
        self.first = None
        self.last = None
        self.flag = False
        self.prev = None

    def helper(self, root):

        if not root:
            return

        self.helper(root.left)

        if self.prev is not None and self.prev.val > root.val:
            if not self.flag:
                self.flag = True
                self.first = self.prev
                self.last = root
            else:
                self.last = root

        self.prev = root
        self.helper(root.right)

    def recoverTree(self, root: TreeNode) -> None:
        """
        Do not return anything, modify root in-place instead.
        """

        if(root is None):
            return
        self.helper(root);
        temp = self.last.val;
        self.last.val = self.first.val;
        self.first.val = temp;